extras/models.py

```diff
@@ -196,3 +197,42 @@ def set_output(self, output, obj=None):
             grouping="output",
             save=False,
         )
+
+
+class ServiceReference(models.Model):
+    owner_type = models.ForeignKey(
+        to=ContentType, on_delete=models.PROTECT, related_name="+"
+    )
+    owner_id = models.PositiveIntegerField()
+    owner = GenericForeignKey(ct_field="owner_type", fk_field="owner_id")
+    prefix = models.CharField(max_length=15, null=True, blank=True)
+    suffix = models.CharField(max_length=15, null=True, blank=True)
+    identifier = models.CharField(
+        max_length=127, null=True, blank=True, db_index=True, unique=True
+    )
+
+    @property
+    def pid(self):
+        return self.id + 100000
```

Comment:
Why `100000` as hard coded value?

Reply:
Just for visual reasons. Starts the reference at 100001 instead of 1.